foreman/data_refinery_foreman/surveyor/array_express.py

```diff
@@ -16,68 +17,111 @@
+    @staticmethod
+    def get_experiment_metadata(experiment_accession_code):
+        experiment_request = requests.get(EXPERIMENTS_URL + experiment_accession_code)
+        parsed_json = experiment_request.json()["experiments"]["experiment"][0]
+
+        experiment = {}
+
+        experiment["name"] = parsed_json["name"]
+        experiment["experiment_accession_code"] = experiment_accession_code
+
+        # If there is more than one arraydesign listed in the experiment
+        # then there is no other way to determine which array was used
+        # for which sample other than looking at the header of the CEL
+        # file. That obviously cannot happen until the CEL file has been
+        # downloaded so we can just mark it as UNKNOWN and let the
+        # downloader inspect the downloaded file to determine the
+        # array then.
+        if len(parsed_json["arraydesign"]) == 0:
+            logger.warn("Experiment %s has no arraydesign listed.", experiment_accession_code)
+            experiment["platform_accession_code"] = "UNKNOWN"
+        elif len(parsed_json["arraydesign"]) > 1:
+            experiment["platform_accession_code"] = "UNKNOWN"
+        else:
+            experiment["platform_accession_code"] = \
+                parsed_json["arraydesign"][0]["accession"]
+
+        experiment["release_date"] = parsed_json["releasedate"]
+
+        if "lastupdatedate" in parsed_json:
+            experiment["last_update_date"] = parsed_json["lastupdatedate"]
+        else:
+            experiment["last_update_date"] = parsed_json["releasedate"]
+
+        return experiment
+
+    def survey(self):
+        experiment_accession_code = (
+            SurveyJobKeyValue
+            .objects
+            .get(survey_job_id=self.survey_job.id,
+                 key__exact="experiment_accession_code")
+            .value
+        )
+
+        logger.info("Surveying experiment with accession code: %s.", experiment_accession_code)
+
+        experiment = self.get_experiment_metadata(experiment_accession_code)
+
+        r = requests.get(SAMPLES_URL.format(experiment_accession_code))
+        samples = r.json()["experiment"]["sample"]
+
+        batches = []
+        for sample in samples:
+            if "file" not in sample:
+                continue
+
+            organism_name = "UNKNOWN"
+            for characteristic in sample["characteristic"]:
+                if characteristic["category"].upper() == "ORGANISM":
+                    organism_name = characteristic["value"].upper()
+
+            if organism_name == "UNKNOWN":
+                logger.error("Sample from experiment %s did not specify the organism name.",
+                             experiment_accession_code)
+                organism_id = 0
+            else:
+                organism_id = Organism.get_id_for_name(organism_name)
+
+            for sample_file in sample["file"]:
+                if sample_file["type"] != "data" and sample_file["name"] is not None:
```

Comment:
Ok this line of code confused me. I can't tell why we would ever want to continue if `sample_file["name"]` is `None` since we use that value later and call split on it. I think it's probably just a bug so I'm going to remove everything from `and` onward.